diff -urNp -x '*.orig' percona-server-5.7.31-34.org/cmake/readline.cmake percona-server-5.7.31-34/cmake/readline.cmake
--- percona-server-5.7.31-34.org/cmake/readline.cmake	2020-08-15 11:23:18.000000000 +0200
+++ percona-server-5.7.31-34/cmake/readline.cmake	2021-05-26 15:30:14.903462364 +0200
@@ -107,23 +107,23 @@ ENDMACRO()
 
 MACRO (FIND_SYSTEM_EDITLINE)
   FIND_PATH(FOUND_EDITLINE_READLINE
-    NAMES editline/readline.h
+    NAMES readline/readline.h
   )
   IF(FOUND_EDITLINE_READLINE)
-    SET(EDITLINE_INCLUDE_DIR "${FOUND_EDITLINE_READLINE}/editline")
+    SET(EDITLINE_INCLUDE_DIR "${FOUND_EDITLINE_READLINE}/readline")
   ELSE()
     # Different path on FreeBSD
     FIND_PATH(FOUND_EDIT_READLINE_READLINE
       NAMES edit/readline/readline.h
     )
     IF(FOUND_EDIT_READLINE_READLINE)
-      SET(EDITLINE_INCLUDE_DIR "${FOUND_EDIT_READLINE_READLINE}/edit/readline")
+      SET(EDITLINE_INCLUDE_DIR "${FOUND_EDIT_READLINE_READLINE}/readline")
     ENDIF()
   ENDIF()
 
   FIND_LIBRARY(EDITLINE_LIBRARY
     NAMES
-    edit
+    readline
   )
   MARK_AS_ADVANCED(EDITLINE_INCLUDE_DIR EDITLINE_LIBRARY)
 
@@ -138,7 +138,7 @@ MACRO (FIND_SYSTEM_EDITLINE)
     LIST(APPEND CMAKE_REQUIRED_LIBRARIES ${EDITLINE_LIBRARY})
     CHECK_CXX_SOURCE_COMPILES("
     #include <stdio.h>
-    #include <readline.h>
+    #include <readline/history.h>
     int main(int argc, char **argv)
     {
        HIST_ENTRY entry;
@@ -148,14 +148,13 @@ MACRO (FIND_SYSTEM_EDITLINE)
 
     CHECK_CXX_SOURCE_COMPILES("
     #include <stdio.h>
-    #include <readline.h>
+    #include <readline/readline.h>
     int main(int argc, char **argv)
     {
-      typedef int MYFunction(const char*, int);
-      MYFunction* myf= rl_completion_entry_function;
-      int res= (myf)(NULL, 0);
-      completion_matches(0,0);
-      return res;
+      rl_compentry_func_t *myf= rl_completion_entry_function;
+      char *res= (myf)(NULL, 0);
+      rl_completion_matches(0,0);
+      return res ? 0 : 1;
     }"
     XLINE_HAVE_COMPLETION_INT)
 
